core/embed/rust/src/ui/model_mercury/component/status_screen.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    time::Duration,
+    ui::{
+        component::{Component, Event, EventCtx, PageMsg},
+        display::{Color, Icon},
+        geometry::{Alignment2D, Rect},
+        shape,
+        shape::Renderer,
+    },
+};
+
+use super::{theme, Swipe, SwipeDirection};
+
+/// Component showing status of an operation. Most typically embedded as a
+/// content of a Frame and showing success (checkmark with a circle around).
+pub struct StatusScreen {
+    area: Rect,
+    icon: Icon,
+    icon_color: Color,
+    circle_color: Color,
+    dismiss_type: DismissType,
+}
+
+enum DismissType {
+    SwipeUp(Swipe),
+    Timeout(Duration), // TODO: handle disappearing StatusScreen
+}
+
+impl StatusScreen {
+    fn new(icon: Icon, icon_color: Color, circle_color: Color, dismiss_style: DismissType) -> Self {
+        Self {
+            area: Rect::zero(),
+            icon,
+            icon_color,
+            circle_color,
+            dismiss_type: dismiss_style,
+        }
+    }
+
+    pub fn new_success() -> Self {
+        Self::new(
+            theme::ICON_SIMPLE_CHECKMARK,
+            theme::GREEN_LIME,
+            theme::GREEN_LIGHT,
+            DismissType::SwipeUp(Swipe::new().up()),
+        )
+    }
+
+    pub fn new_neutral() -> Self {
+        Self::new(
+            theme::ICON_SIMPLE_CHECKMARK,
+            theme::GREY_EXTRA_LIGHT,
+            theme::GREY_DARK,
+            DismissType::SwipeUp(Swipe::new().up()),
+        )
+    }
+}
+
+impl Component for StatusScreen {
+    type Msg = PageMsg<()>;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.area = bounds;
+        match self.dismiss_type {
+            DismissType::SwipeUp(ref mut swipe) => {
+                swipe.place(bounds);
+            }
+            DismissType::Timeout(_) => {
+                // TODO: handle timeout or scrub the idea?
```

Comment:
My suggestion would be to embed a `struct Timeout` and handle it there.

Reply:
Ok, the first version is here: https://github.com/trezor/trezor-firmware/pull/3740/commits/317ed644799691ccbb6798c7b3cf5ba85eab8f78 
Very simple ctor supplied, for now the timeout is not parametrizable. Let's see what needs arise later.